### PR TITLE
quaternion trigonometry

### PR DESCRIPTION
**Pull request by adamnemecek**
The test values are based on qtfm.

### CODE REVIEW

**Comment by sebcrozet** [comment]
@adamnemecek Thank you for those! The tests should be doc-tests instead of completely separate functions (see [this](https://github.com/rustsim/nalgebra/blob/master/src/geometry/quaternion.rs#L114-L121) for an example of doc-test).

**Comment by Ralith** [comment]
Out of curiosity, what are these used for?

File: src/geometry/quaternion_construction.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ use base::dimension::U3;
 use base::storage::Storage;
 #[cfg(feature = "arbitrary")]
 use base::Vector3;
```
[the remark this answers is not in the document]
Comment (adamnemecek) [reply src/geometry/quaternion_construction.rs]: Ok fixed.

**Comment by adamnemecek** [comment]
@Ralith this question is very general. I have some code that can also be expressed in terms of quaternion trigonometry. I use these functions to make sure that my implementation is correct.

**Comment by sebcrozet** [comment]
Thanks!

**Pull request merged**